### ISSUE TITLE
this.res._headers has changed in node master

### ISSUE THREAD

**Issue by ilkkao** [koajs/koa#903]
Just compiled Node from the master branch and tried to use koa. It failed because the property values in `this.res._headers` are now arrays instead of strings.

```
$ node
> http = require('http');
> resp = new http.ServerResponse({});
> resp.setHeader('Content-Type', 'foo');
> result = resp._headers
```

In Node 7.x, `result` is `{ 'content-type': 'foo' }`

In Node master (pre-8.0), `result` is `{ 'content-type': [ 'Content-Type', 'foo' ] }`

The change was made here: https://github.com/nodejs/node/commit/c00e4adbb4c6dd5bfbf22a3efcecb63c56958339#diff-286202fdbdd74ede6f5f5334b6176b5cR406 (Node PR: https://github.com/nodejs/node/pull/10558)

@mscdex noticed that it broke express and provided a fix https://github.com/jshttp/fresh/pull/20/files Something similar would be needed for koa.

I'll open this issue to make sure this issue is solved for koa in some way.

**Comment by mscdex** [comment]
I would probably wait until https://github.com/nodejs/node/pull/10805 lands, which will provide more public API functions around accessing HTTP headers (there is a separate PR to deprecate `_headers`).

In some cases though using functions such as the existing `getHeader()` is enough for some use cases.

**Comment by ilkkao** [comment]
I can prepare a PR when `response.getHeaders` from nodejs/node#10805 has landed

**Comment by mscdex** [comment]
Just a heads up, the PR has landed now in https://github.com/nodejs/node/commit/3e6f1032a4fdb8ca7fba02c7d2103fba68c0ee1f.

**Comment by ilkkao** [comment]
Koa PR open: https://github.com/koajs/koa/pull/907

**Comment by ilkkao** [comment]
Code in the master branch still uses `this.res._headers` directly. Shouldn't the fix go to master too?

**Comment by jonathanong** [comment]
@ilkkao yes, that would be great!

**Comment by ilkkao** [comment]
I'll send a PR in few days

**Comment by JasonBoy** [comment]
@ilkkao `getHeaders` and `getHeaderNames` are added in node-v7.7, the comment `Node < 8` in #907 could be updated to `Node < 7.7`?

**Comment by jonathanong** [comment]
released in 2.1.0 and 1.3.0 thank you @ilkkao !!
